src/bun.js/ConsoleObject.zig

```diff
@@ -72,6 +72,9 @@ pub const MessageType = enum(u32) {
 var stderr_mutex: bun.Lock = bun.Lock.init();
 var stdout_mutex: bun.Lock = bun.Lock.init();
 
+threadlocal var stderr_lock_count: u16 = 0;
```

Comment:
This can be a boolean and then a stack-local boolean and then there's no risk of stack overflow

Reply:
Can it be? 
I haven't tried it out but you should be able to get multiple levels of recursion. 
For example when adding another MyMap to https://github.com/oven-sh/bun/pull/8213/files#diff-aa4ea7eac69e2fd1b42ca19ac25fbc56508a271a55a0ac6d584c5ff5278bd905R10 and printing that as well. 

We can bump the u16 to something else. I felt pretty comfortable with 65k logs but who knows lol.

Reply:
Sorry for the edits. Was confused for a sec. I reverted to the original message.